docs/install.rst

```diff
@@ -14,153 +14,148 @@ Once installation is complete, you can run :ref:`mitmproxy`, :ref:`mitmdump` or
-supports HTTP/2. Of course, you can also install mitmproxy from source if you
-prefer that (see :ref:`install-advanced`).
-
-.. _install-advanced:
+The recommended way to install mitmproxy on a recent Linux distribution (with
+Python 3.5 or higher) is to use pip3:
 
-Advanced Installation
----------------------
+Please make sure to install Python 3.5 (or higher) and pip3 for your
+distribtion. If your distribution does not provide a suitable Python version,
+you can use `pyenv`_ to get a recent Python environment.
 
-.. _install-docker:
-
-Docker Images
-^^^^^^^^^^^^^
-
-You can also use the official mitmproxy images from `DockerHub`_. That being
-said, our portable binaries are just as easy to install and even easier to use. 😊
+.. code:: bash
 
+    sudo apt-get install python3-pip # Debian 8 or higher, Ubuntu 16.04 or higher
+    sudo dnf install python3-pip # Fedora 24 or higher
+    sudo pacman -S python-pip # Arch Linux
 
-.. _install-arch:
+Please make sure to upgrade pip3 itself:
 
-Installation on Arch Linux
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. code:: bash
 
-mitmproxy has been added into the [community] repository. Use pacman to install it:
+    sudo pip3 install -U pip
 
->>> sudo pacman -S mitmproxy
+Now you can install mitmproxy via pip3:
 
+.. code:: bash
 
-.. _install-source-ubuntu:
+    sudo pip3 install mitmproxy
 
-Installation from Source on Ubuntu
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you are using a legacy distribution without support for Python 3.5 or higher,
+we provide self-contained pre-built binaries on our `releases`_ page.
 
-Ubuntu comes with Python but we need to install pip3, python3-dev and several
-libraries. This was tested on a fully patched installation of Ubuntu 16.04.
 
-.. code:: bash
+.. _install-advanced:
 
-   sudo apt-get install python3-pip
-   sudo pip3 install mitmproxy
+Advanced Installation
+---------------------
 
-On older Ubuntu versions, e.g., **12.04** and **14.04**, you may need to install
-a newer version of Python. mitmproxy requires Python 3.5 or higher. Please take
-a look at pyenv_. Make sure to have an up-to-date version of pip by running
-``pip3 install -U pip``.
+.. _binary-packages:
 
+Self-contained Pre-built Binary Packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. _install-source-fedora:
+For some platforms we provide pre-built binary packages containing ready-to-run
+executables. This includes a self-contained Python 3 environment, a recent
+OpenSSL that support ALPN and HTTP/2, and other dependencies that would
+otherwise we cumbersome to compile and install.
 
-Installation from Source on Fedora
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Please be advised that we do not updates these binaries after releasing a new
+mitmproxy version. This means we do not include security-related updates of our
+dependencies in already released mitmproxy versions. If there is a severe issue,
+we might consider releasing a bugfix release of mitmproxy and corresponding
+binary packages.
```

Comment:
+1 on mentioning this, but this also applies to the non-prebuilt installations as we pin e.g. the cryptography version. Maybe we should generalize it to a simple "Only the latest release is guaranteed to receive security updates".

Reply:
We don't pin cryptography to a specific version, only to a `v2`: https://github.com/mitmproxy/mitmproxy/blob/781369a3269c05d40fa8d37cfc0eae0401558d8d/setup.py#L67
Or what do you mean with this?

Reply:
Once cryptography releases their next minor release (2.1), they won't provide security updates for 2.0 anymore, so we end up with pretty much the same situation - old releases will have old OpenSSL versions. Of course there are small advantanges (e.g. certifi is unbounded in setup.py), but the issue generally remains.